### PR TITLE
Django requests

### PR DESCRIPTION
**Pull request by peterLERNSAX**

### CODE REVIEW

File: api/lagerdb/models/models.py [hunk 1]
```diff
@@ -15,8 +15,46 @@ class Layout(Base):
...
+
+    __tablename__ = "offer"
+
+    id = Column(Integer,primary_key=True)
+    customer = Column(Integer,ForeignKey("customer.id"), nullable=True)
+    offer_file = Column(Integer, ForeignKey("offerfile.id"), nullable=True)
+    layout = Column(Integer,ForeignKey("layout.id"),nullable=True)
+    description = Column(String,nullable=True)
+
+
```
Comment (jonasLERNSAX) [inline api/lagerdb/models/models.py:34]: 2 Leerzeilen?

File: api/lagerdb/models/models.py [hunk 2]
```diff
@@ -15,8 +15,46 @@ class Layout(Base):
...
+    __tablename__ = "customer"
+
+    id = Column(Integer,primary_key=True)
+    firstname = Column(String,nullable=False)
+    sirname = Column(String,nullable=False)
+    email = Column(String,nullable=False)
+    company_name = Column(String,nullable=True)
+
+
```
Comment (jonasLERNSAX) [inline api/lagerdb/models/models.py:60]: 2 Leerzeilen?

File: api/main.py [hunk 3]
```diff
@@ -39,6 +39,37 @@ def return_403():
...
+async def get_all_hardware(perms:Permission):
+    """
+    Route for getting all hardware from db
+    """
+    if not check_permission(perms,4):
+        return return_403()
+    hardware = db_manager.show_all_hardware()
+    return {"response":hardware}
+
+
```
Comment (jonasLERNSAX) [inline api/main.py:72]: 2 Leerzeilen?

File: softwaredd/usermanagement/models.py [hunk 4]
```diff
@@ -10,4 +10,14 @@ class Employee(User):
...
+    def get_permission_dict(self)->dict[str,bool]:
+        """returns permission dict"""
+        return {
+            "usermanagement":self.perm_usermanagement,
+            "layout":self.perm_layout,
+            "database":self.perm_database,
+            "offer":self.perm_offer,
+            "offer_file":self.perm_offer_file,
+                }
+
```
Comment (jonasLERNSAX) [inline softwaredd/usermanagement/models.py:22]: 2 Leerzeilen?

File: softwaredd/usermanagement/templates/usermanagement/list_customer.html [hunk 5]
```diff
@@ -0,0 +1,7 @@
+{% extends "usermanagement/index_webserver.html" %}
+{% load static %}
+{% block content %}
+{%for layout in content%}
+{{layout}}
+{%endfor%}
+{% endblock %}
```
Comment (jonasLERNSAX) [inline softwaredd/usermanagement/templates/usermanagement/list_customer.html:7]: missing last line

File: softwaredd/usermanagement/templates/usermanagement/list_hardware.html [hunk 6]
```diff
@@ -0,0 +1,7 @@
+{% extends "usermanagement/index_webserver.html" %}
+{% load static %}
+{% block content %}
+{%for layout in content%}
+{{layout}}
+{%endfor%}
+{% endblock %}
```
Comment (jonasLERNSAX) [inline softwaredd/usermanagement/templates/usermanagement/list_hardware.html:7]: missing last line

File: softwaredd/usermanagement/templates/usermanagement/list_layout.html [hunk 7]
```diff
@@ -0,0 +1,7 @@
+{% extends "usermanagement/index_webserver.html" %}
+{% load static %}
+{% block content %}
+{%for layout in content%}
+{{layout}}
+{%endfor%}
+{% endblock %}
```
Comment (jonasLERNSAX) [inline softwaredd/usermanagement/templates/usermanagement/list_layout.html:7]: missing new line at eof

File: softwaredd/usermanagement/templates/usermanagement/list_offer.html [hunk 8]
```diff
@@ -0,0 +1,7 @@
+{% extends "usermanagement/index_webserver.html" %}
+{% load static %}
+{% block content %}
+{%for offer in content%}
+{{offer}}
+{%endfor%}
+{% endblock %}
```
Comment (jonasLERNSAX) [inline softwaredd/usermanagement/templates/usermanagement/list_offer.html:7]: missing new line at eof

File: softwaredd/usermanagement/templates/usermanagement/list_offers.html [hunk 9]
```diff
@@ -0,0 +1,7 @@
+{% extends "usermanagement/index_webserver.html" %}
+{% load static %}
+{% block content %}
+{%for layout in content%}
+{{layout}}
+{%endfor%}
+{% endblock %}
```
Comment (jonasLERNSAX) [inline softwaredd/usermanagement/templates/usermanagement/list_offers.html:7]: missing new line at eof

File: softwaredd/usermanagement/urls.py [hunk 10]
```diff
@@ -26,4 +26,9 @@
     path("use-case/", views.UseCaseView.as_view(), name="use-case-view"),
     path("copyright/", views.CopyrightView.as_view(), name="copyright-view"),
     path("webserver/", views.WebserverView.as_view(), name="webserver-view"),
+    path("grundrisse/",views.ListLayoutView.as_view(),name="list-layout-view"),
+    path("angebotsdateien/",views.ListOfferView.as_view(),name="list-offer-view"),
+    path("kunden/",views.ListCustomerView.as_view(),name="list-customer-view"),
+    path("hardware/",views.ListHardwareView.as_view(),name="list-hardware-view"),
+    path("angebote/",views.ListOffersView.as_view(),name="list-offers-view")
 ]
```
Comment (jonasLERNSAX) [inline softwaredd/usermanagement/urls.py:34]: missing new line at eof

File: softwaredd/usermanagement/views.py [hunk 11]
```diff
@@ -169,7 +171,70 @@ def employee_authentication(user: Employee, check_value: str) -> bool:
...
+        """get"""
+        if not request.user.is_authenticated:
+            messages.warning(request,"Nicht Authentifiziert!")
+            return redirect("index-view")
+        url = API_URL+"/dbApi/v1/post/offer/all/"
+        employee:Employee = Employee.objects.get(pk=request.user.pk)
+        content:Response=requests.post(url,json=employee.get_permission_dict())
+        content_dict = (json.loads(content.content))
+        return render(request,"usermanagement/list_offers.html",{"content":content_dict["response"]})
+
```
Comment (jonasLERNSAX) [inline softwaredd/usermanagement/views.py:224]: 2 Leerzeilen

File: softwaredd/usermanagement/urls.py (hunk 10 above)
Comment (peterLERNSAX) [reply softwaredd/usermanagement/urls.py:34]: FALSCH

File: softwaredd/usermanagement/views.py (hunk 11 above)
Comment (peterLERNSAX) [reply softwaredd/usermanagement/views.py:224]: FALSCH

**Pull request merged**